LibertyDSNP/example-client
# Feature/setup redux store 177598915

Purpose
---------------
As a developer we want to use redux to store our state globally to simplify the applications state management and prop passing.


Solution
---------------
Implement a redux store and fix the existing code to use that store instead of default react state.
with @claireolmstead 


Change summary
---------------
* Added a redux store using the redux & react-redux & redux-toolkit modules
* Created global state with user, graph, profile, and feed store slices.
* Removed react state passing and simplified components.
* Fixed tests to use redux by implementing the redux-mock-store module.


Steps to Verify
----------------
1. Log in
2. Current state data appears in profile block
3. If Redux extension installed check state in Elements inspector
